public/map-controller/map-controller.js

```diff
@@ -129,11 +131,61 @@ var MapController = P({
     };
   },
 
+  build_epi_overlay_layer: function(epi_data) {
+    var layer_group = L.layerGroup();
+    _.forEach(epi_data.region_case_data, (function(case_data, region_code) {
+      var latlng = this.region_code_to_latlng[region_code];
+      // TODO(jetpack): scale by relative admin size for the country, or something?
+      var radius_meters = 20000 * this.map_coloring.case_data_to_severity(case_data);
+      var circle = L.circle(latlng, radius_meters, {opacity: 0.9, fillOpacity: 0.7});
+
+      var circle_popup = L.popup(this.popup_options);
+      var region_name = this.region_details.get_region_properties(region_code).name;
+      circle_popup.setContent('<b>' + region_name + '</b><br/>' +
+                              this.map_coloring.case_data_to_html_string(case_data));
+      var map = this.map;
+      // TODO(jetpack): clicks on the circle should behave the same as clicks on
+      // the region.
+      circle.on({
+        mousemove: function(e) {
+          circle_popup.setLatLng(e.latlng);
+          map.openPopup(circle_popup);
+        },
+        mouseout: function() { map.closePopup(circle_popup); }
+      });
+
+      layer_group.addLayer(circle);
+    }).bind(this));
+    return layer_group;
+  },
+
   redraw: function() {
     var style_fcn = this.get_region_style_fcn();
     _.forEach(this.regions_layers, function(layer) {
```

Comment:
added a note to https://github.com/unicef/carrotjuice2/issues/17 -- in the future, we might want to only call `setStyle` when the data actually changes. for example, if you toggle an overlay layer, probably good not to redraw the base.